homeassistant/components/media_player/kodi.py

```diff
@@ -24,16 +26,19 @@
 
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
-    """ Sets up the kodi platform. """
+    """ Sets up the kodi media player platform. """
 
-    add_devices([
-        KodiDevice(
-            config.get('name', 'Kodi'),
-            config.get('url'),
-            auth=(
-                config.get('user', ''),
-                config.get('password', ''))),
-    ])
+    if 'url' in config:
```

Comment:
If you're using discovery, you can see that people are using the legacy format by testing if `discovery_info is None`. This means that they added the media player directly to their config instead of using the main `kodi` platform.